### PR TITLE
Cherry-pick arabic text to release branch v3.6.0

### PR DESCRIPTION
**Pull request by fabian-guerra**
Cherry-picks https://github.com/mapbox/mapbox-gl-native/pull/9062 and https://github.com/mapbox/mapbox-gl-native/pull/9031 to release-ios-v3.6.0-android-5.1.0

cc/ @mapbox/android

### CODE REVIEW

**Comment by 1ec5** [comment]
This PR says it brings in #9031, but I don’t see 77b24bc0d3ddfda494cf1a7ce4090ff3a79324e4 in here.

**Comment by ChrisLoer** [comment]
This PR does however include the fix for issue #8982 -- was that intentional? I'd characterize that as a low-impact bug with a low-risk fix.

**Comment by fabian-guerra** [comment]
@1ec5 @ChrisLoer it wan't intention I grabbed wrong SHA that's why it was included. I'm removing and adding the right commits.

File: src/mbgl/tile/geometry_tile.hpp [hunk 1]
```diff
@@ -87,6 +88,9 @@ class GeometryTile : public Tile, public GlyphRequestor, IconRequestor {
     }
 
 private:
+    void markObsolete();
```
Comment (ChrisLoer) [inline src/mbgl/tile/geometry_tile.hpp]: This `markObsolete` looks like it snuck in from `master`, it's not used on this branch.

**Pull request merged**